pandas/_libs/algos_groupby_helper.pxi.in

```diff
@@ -55,37 +56,65 @@ def group_add_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
 
     with nogil:
 
-        if K > 1:
+        if skipna == False:
+            if K > 1:
 
-            for i in range(N):
-                lab = labels[i]
-                if lab < 0:
-                    continue
-
-                counts[lab] += 1
-                for j in range(K):
-                    val = values[i, j]
+                for i in range(N):
```

Comment:
Okay I'll revert it to the usual single block. Thought it was better not to do one extra check every iteration. :stuck_out_tongue: Thanks for the feedback.

So do you suggest handling the datetimes in the cython routines just like `cummin` using the `is_datetimelike` arg?